### PR TITLE
CT-2364 Introduce Offender SAR Monthly report download option

### PR DESCRIPTION
**Pull request by mseedat-moj**
## Description
Allows a user to download a monthly KPI report based on existing SAR/FOI monthly reports. Does not include any bespoke amendments for Offender SARs (i.e. initial plumbing & download functionality only).

## Notes
- Rules for `business_unit_in_time?` will require consideration
- Report output uses existing `trigger` terminology, reconsider
- User restrictions not implemented
 
## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
 
### Screenshots
Screen Shot 2019-09-04 at 11 02 31

 
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2364
 
### Deployment
1. Database migration required
2. Data Migration required
 
### Manual testing instructions
1. Log in as any user
2. Performance -> Offender Sars section. 
3. Select 'download' should download an Excel file

### CODE REVIEW

**Comment by mseedat-moj** [comment]
@Polsonby Review is required but this PR cannot be merged due to the `@todo` in `offender#business_unit_already_late?`. Agreed with Alex

File: app/views/stats/index.html.slim [hunk 1]
```diff
@@ -15,10 +15,17 @@
...
+    correspondence_type: 'Standard FOI reports',                               \
+    reports: @reports[:foi],                                                   \
+  }                                                                            
 
-= render partial: 'stats/report',
-        locals: { correspondence_type: 'Standard SAR reports',
-                reports: @sar_reports }
+= render partial: 'stats/report', locals: {                                    \
+    correspondence_type: 'Standard SAR reports',                               \
+    reports: @reports[:sar],                                                   \
```
Comment (Polsonby) [inline app/views/stats/index.html.slim]: We don't need backslashes if the line ends with a comma, so I think only the first one is required

File: app/views/stats/index.html.slim [hunk 2]
```diff
@@ -15,10 +15,17 @@
...
+  }                                                                            
+
+= render partial: 'stats/report', locals: {                                    \
+    correspondence_type: 'Offender SAR reports',                               \
+    reports: @reports[:offender_sar],                                          \
```
Comment (Polsonby) [inline app/views/stats/index.html.slim]: Same here?

**Review by Polsonby — APPROVED** [review]
LGTM :) 👍

**Pull request merged**